src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -479,5 +481,33 @@ private static void BlockCyclicAggregateTypeReferences(SemanticModel model, IDia
                 _ => DiagnosticBuilder.ForPosition(kvp.Key.DeclaringType.Name).CyclicType(kvp.Value.Select(s => s.Name)),
             }));
         }
+
+        private static ImmutableDictionary<ParameterAssignmentSymbol, JToken> CalculateParameterAssignments(SemanticModel model, IDiagnosticWriter diagnostics)
+        {
+            var generated = new Dictionary<ParameterAssignmentSymbol, JToken>();
+            var evaluator = new ParameterAssignmentEvaluator(model);
+            foreach (var parameter in model.Root.ParameterAssignments)
+            {
+                if (model.GetTypeInfo(parameter.DeclaringSyntax) is ErrorType)
+                {
+                    // no point evaluating if we're already reporting an error
+                    continue;
+                }
+
+                // We may emit duplicate errors here - type checking will also execute some ARM functions and generate errors
+                // This is something we should improve before the first release.
```

Comment:
One option here may be to use the type of the parameter assignment instead of re-evaluating. I'm not sure if it covers all cases, but if `model.GetTypeInfo(parameter.DeclaringSyntax)` returns a type for which `TypeHelper.IsLiteralType` returns `true`, it can be converted directly to a JToken.

Reply:
I started out with a similar approach - relying purely on type analysis to catch issues, but found that there were cases where the conversion wasn't being blocked (leading to a compilation failure). This definitely feels like a viable approach if we're able to guarantee we're always going to get either an error or a literal type.